backuprestore/restore.py

```diff
@@ -33,7 +34,97 @@ def main():
             iaas_client.exit(
                 'Could not find the persistent volume attached to this instance.')
 
-        if landscape != 'Aws' and landscape != 'Azure' and landscape != 'Gcp':
+        if landscape == 'Aws' or landscape == 'Azure' or landscape == 'Gcp':
+            try:
+                # get sanpshot id from service metadata stored in blobstore
+                is_snapshot_present = iaas_client.download_from_blobstore(
+                    '{}/{}'.format(backup_guid, metadata_files_name), metadata_files_path)
+            except Exception as error:
+                iaas_client.logger.error(error)
+                is_snapshot_present = False
+
+            if (not is_snapshot_present) and landscape != 'Azure':
```

Comment:
Just a simple error handling, I guess!
Suppose, you are attempting a restore from a snapshot in case of AWS or GCP and the `metadata.json` file itself is deleted (somehow), then you can't proceed with the restore, right?
This check will catch that. In case of Azure, this mightn't point exactly to an error, since, it could be blob based too.

Reply:
ok, got it , thanks !